media_service/tests/test_iiif.py

```diff
@@ -71,3 +91,27 @@ def test_manifest_attributes(self):
             self.assertTrue(key in md, "manifest should have a %s attribute" % key)
             self.assertEqual(md[key], expected_attr[key])
     
+    def test_manifest_has_unique_canvas_ids(self):
+        # create a new list of images with duplicates of first and last image
+        images = self.get_images_list()
+        for image_id in (images[0]['id'], images[-1]['id']):
+            self.duplicate_image(images, image_id, num_duplicates=3)
```

Comment:
Good idea to verify pre-condition for test. I'd rather not hardcode the number of duplicates, so I'll see if I can make that change.